xmtp_proto/src/types/global_cursor.rs

```diff
@@ -8,25 +8,33 @@ use crate::{
 use core::fmt;
 use serde::{Deserialize, Serialize};
 use std::{
-    collections::HashMap,
+    collections::{BTreeMap, HashMap},
     fmt::Write,
     ops::{Deref, DerefMut},
 };
 use xmtp_configuration::Originators;
 
-/// a cursor which represents the position across many nodes in the network
+/// a cursor backed by a [`BTreeMap`].
+/// represents the position across many nodes in the network
 /// a.k.a vector clock
-#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[derive(Default, Debug, Clone, Hash, Serialize, Deserialize, PartialEq, Eq)]
 pub struct GlobalCursor {
-    inner: HashMap<OriginatorId, SequenceId>,
+    inner: BTreeMap<OriginatorId, SequenceId>,
```

Comment:
changed the backing data structure from a `HashMap` to a `BTreeMap`. The `BTreeMap` makes more sense to me, since we can order the map by keys and also derive `Hash` on it. the semantics of inserting/"contains" etc. is unchanged